src/compiler/types.ts

```diff
@@ -5288,6 +5289,8 @@ export interface TypeChecker {
     tryGetThisTypeAt(node: Node, includeGlobalThis?: boolean, container?: ThisContainer): Type | undefined;
     /** @internal */ getTypeArgumentConstraint(node: TypeNode): Type | undefined;
 
+    filterType(type: Type, f: (t: Type) => boolean): Type;

```

Comment:
Eek, definitely not sure about putting this in the exported interface, let alone the public API...

Reply:
Okay, I will find a way to solve the completion internally without creating more public api.